qa/tasks/vstart_runner.py

```diff
@@ -325,9 +325,14 @@ def _perform_checks_and_return_list_of_args(self, args, omit_sudo):
 
         return args
 
-    def run(self, args, check_status=True, wait=True,
-            stdout=None, stderr=None, cwd=None, stdin=None,
-            logger=None, label=None, env=None, timeout=None, omit_sudo=False):
+    # Wrapper to keep the interface exactly same as that of
+    # teuthology.remote.run.
+    def run(self, **kwargs):
+        return self.cmdrunner(**kwargs)
+
+    def cmdrunner(self, args, check_status=True, wait=True, stdout=None,
```

Comment:
if nobody is using `cmdrunner()`, why bother having it?

Reply:
Do you mean that you want to merge the two or just get rid of `cmdrunner()` and all the underlying code? I wanted the interfaces to be be exactly same as https://github.com/ceph/teuthology/blob/master/teuthology/orchestra/remote.py#L188 and https://github.com/ceph/teuthology/blob/master/teuthology/orchestra/run.py#L366.

Reply:
yes. i cannot find a caller other than `run()`. so i am wondering why can't we just ditch it.

Reply:
It's not about having a caller for the method but about keep the interfaces *exactly* same as that of teuthology's run method so that I, or anybody else, doesn't get a test failure rising just out of a difference in how arguments were passed. 

Here's the first run method in teuthology that get's called by test code -
```
   def run(self, **kwargs):
        """
        This calls `orchestra.run.run` with our SSH client.
        TODO refactor to move run.run here?
        """
```
I've posted the link to it before too - https://github.com/ceph/teuthology/blob/master/teuthology/orchestra/remote.py#L188

And here's other run() method in teuthology called by the former run method -
```
def run(
    client, args,
    stdin=None, stdout=None, stderr=None,
    logger=None,
    check_status=True,
    wait=True,
    name=None,
    label=None,
    timeout=None,
    cwd=None,
    # omit_sudo is used by vstart_runner.py
    omit_sudo=False
):
```
https://github.com/ceph/teuthology/blob/master/teuthology/orchestra/run.py#L366

After this PR merges, the roundtrip, at least for this matter, would be prevented.

Reply:
@rishabh-d-dave what i suggest is to remove `cmdrunner()` not to remove `run()`.

Reply:
@tchaikov Hopefully, this brings some clarity to our conversation - https://tracker.ceph.com/issues/44117#note-3